PurchasesTests/Mocks/MockSKProduct.swift

```diff
@@ -4,16 +4,54 @@
 //
 
 class MockSKProduct: SKProduct {
```

Comment:
we had two separate mocks for SKProduct, `MockSKProduct` and `MockProduct`. 
I unified them, updated tests and added a whole bunch of fields / mock stuff